tests/testthat/test_schematic_rest_api.R

```diff
@@ -73,74 +160,304 @@ test_that("model_submit successfully uploads to synapse", {
+  )
+  
+  # bad PAT
+  expect_error(
+    storage_dataset_files(
+      url = url,
+      asset_view = "syn23643253",
+      dataset_id = "syn23643250",
+      access_token = paste0(Sys.getenv("SYNAPSE_PAT"), "1234"))
+  )
 })
 
 test_that("model_component_requirements returns list of required components", {
   skip_it()
-  good <- model_component_requirements(url=file.path(schematic_url, "v1/model/component-requirements"),
-                                           schema_url="https://raw.githubusercontent.com/ncihtan/data-models/main/HTAN.model.jsonld",
-                                           source_component="Patient",
-                                           as_graph = FALSE)
-  expect_equal(length(good), 8L)
   
-  expect_error(model_component_requirements(url=file.path(schematic_url, "v1/model/component-requirements"),
-                                       schema_url="https://aaaabad.url.jsonld",
-                                       source_component="Patient",
-                                       as_graph = FALSE))
+  # Successful call
+  reqs <- model_component_requirements(
+    url = file.path(schematic_url, "v1/model/component-requirements"),
+    schema_url = "https://raw.githubusercontent.com/ncihtan/data-models/main/HTAN.model.jsonld",
+    source_component = "Patient",
+    as_graph = FALSE)
+  
+  expect_equal(length(reqs), 8L)
+  
+  # Filepath for url is errant
+  # returns an empty list()
+  reqs <- model_component_requirements(
+    url = file.path(schematic_url, "v1/model/component-requirements-badurl"),
+    schema_url = "https://raw.githubusercontent.com/ncihtan/data-models/main/HTAN.model.jsonld",
+    source_component = "Patient",
+    as_graph = FALSE)
+  
+  expect_equal(length(reqs), 0L)
   
+  reqs <- model_component_requirements(
+    url = file.path(schematic_url, "v1/model/component-requirements"),
+    schema_url = "https://raw.githubusercontent.com/ncihtan/data-models/main/HTAN.model.jsonld",
+    source_component = "Patient",
+    as_graph = TRUE)
+  
+  #FIXME: Does this output make sense?
+  expect_equal(length(reqs), 7L)
 })
 
-# test_that("manifest_download returns a csv.", {
-#   skip_it()
-#  csv <- manifest_download(url=file.path(schematic_url, "v1/manifest/download"),
-#                           manifest_id="syn51078535",
-#                           access_token=Sys.getenv("SYNAPSE_PAT"))
-#   exp <- setNames(c("BulkRNA-seqAssay", "CSV/TSV", "Sample_A", "GRCm38", NA, 2022L, "syn28278954"),
-#     c("Component", "File Format", "Filename", "Genome Build", "Genome FASTA", "Sample ID", "entityId"))
-#   expect_equal(unlist(csv), exp)
-# })
+test_that("model_component_requirements errors as expected", {
+  
+  # bad schema url
+  expect_error(
+    model_component_requirements(
+      url=file.path(schematic_url, "v1/model/component-requirements"),
+      schema_url="https://aaaabad.url.jsonld",
+      source_component="Patient",
+      as_graph = FALSE)
+    )
+  
+  # FIXME
```

Comment:
@afwillia - just wanted to make sure I flagged some notes about behavior I encountered that could probably use some error handling. Not sure how much of a priority this work is though.